### ISSUE TITLE
JsonLayout attribute includeMdlc is not working in file target in  async wrapper

### ISSUE THREAD

**Issue by Sumoren** [NLog/NLog#3765]
**NLog version**: 4.6.8

**Platform**: .Net 4.5 / .NET Core 3.1

**Current NLog config**

```xml
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

  <targets>
    <target name="logconsole" xsi:type="Console" />
    
    <target
      xsi:type="file"
      name="logfile"
      deleteOldFileOnStartup="true"
      fileName="struct_logs.json">
      <layout xsi:type="JsonLayout">
        <attribute name="time" layout="${longdate}" />
        <attribute name="level" layout="${level:upperCase=true}"/>
        <attribute name="message" layout="${message}" />
        <attribute name="eventProperties" encode="false" >
          <layout type='JsonLayout' includeAllProperties="true"  maxRecursionLimit="2" renderEmptyObject="false"/>
        </attribute>
        <attribute name="scope" encode="false" >
          <layout type='JsonLayout' includeMdlc="true"  maxRecursionLimit="2" renderEmptyObject="false"/>
        </attribute>
      </layout>
    </target>

    <target name="asynclogfile" xsi:type="AsyncWrapper" queueLimit="5000" overflowAction="Discard">
      <target
        xsi:type="file"
        deleteOldFileOnStartup="true"
        fileName="async_struct_logs.json">
        <layout xsi:type="JsonLayout">
          <attribute name="time" layout="${longdate}" />
          <attribute name="level" layout="${level:upperCase=true}"/>
          <attribute name="message" layout="${message}" />
          <attribute name="eventProperties" encode="false" >
            <layout type='JsonLayout' includeAllProperties="true"  maxRecursionLimit="2" renderEmptyObject="false"/>
          </attribute>
          <attribute name="scope" encode="false" >
            <layout type='JsonLayout' includeMdlc="true"  maxRecursionLimit="2" renderEmptyObject="false"/>
          </attribute>
          <!-- Uncomment this line to 'fix' includeMdlc -->
          <!--<attribute name="ExplicitScopeKey" layout="${mdlc:ScopeKey}" />-->
        </layout>
      </target>
    </target>
  </targets>

  <rules>
      <logger name="*" minlevel="Debug" writeTo="logfile" />
      <logger name="*" minlevel="Debug" writeTo="asynclogfile" />
      <logger name="*" minlevel="Debug" writeTo="logconsole" />
  </rules>
</nlog>
```

**Code for testing**
```c#
using NLog;

namespace TestLogScope
{
    class Program
    {
        static void Main(string[] args)
        {
            var logger = LogManager.GetCurrentClassLogger();

            logger.Info("Log before scope");

            using (MappedDiagnosticsLogicalContext.SetScoped("ScopeKey", 52))
            {
                logger.Info("Log inside Mdlc scope {Param}", 79);
            }
        }
    }
}
```
In the `logfile` target the layout `<layout type='JsonLayout' includeMdlc="true"  maxRecursionLimit="2" renderEmptyObject="false"/>` output correctly the scoped MDLC:
```
{ "time": "2020-01-17 13:32:44.7498", "level": "INFO", "message": "Log before scope" }
{ "time": "2020-01-17 13:32:44.7840", "level": "INFO", "message": "Log inside Mdlc scope 79", "eventProperties": { "Param": 79 }, "scope": { "ScopeKey": 52 } }
```
But the scope data is not output in the `asynclogfile` target even if the config is identical:
```
{ "time": "2020-01-17 13:32:44.7498", "level": "INFO", "message": "Log before scope" }
{ "time": "2020-01-17 13:32:44.7840", "level": "INFO", "message": "Log inside Mdlc scope 79", "eventProperties": { "Param": 79 } }
```
If you explicitly use the `${mdlc}` layout renderer by uncommenting the line `<attribute name="ExplicitScopeKey" layout="${mdlc:ScopeKey}" />` in the config then it *fix* the `JsonLayout` with `includeMdlc` in addition to output the `ExplicitScopeKey` attribute:
```
{ "time": "2020-01-17 13:50:00.4026", "level": "INFO", "message": "Log before scope" }
{ "time": "2020-01-17 13:50:00.4397", "level": "INFO", "message": "Log inside Mdlc scope 79", "eventProperties": { "Param": 79 }, "scope": { "ScopeKey": 52 }, "ExplicitScopeKey": "52" }

```

**Comment by snakefoot** [comment]
@Sumoren Thank you for reporting this. And great job with the reproducible sample code.

Have created #3765 for resolving this in NLog 4.7. Will probably steal your code for the unit-tests.

**Comment by snakefoot** [comment]
Closed by #3766
